projects/RabbitMQ.Client/Impl/RabbitMQActivitySource.cs

```diff
@@ -63,14 +66,14 @@ public static class RabbitMQActivitySource
 
             Activity? activity = linkedContext == default
                 ? s_publisherSource.StartRabbitMQActivity(
-                    UseRoutingKeyAsOperationName ? $"{routingKey} publish" : "publish",
+                    UseRoutingKeyAsOperationName ? $"{routingKey} {MessagingOperationTypeSend}" : MessagingOperationTypeSend,
```

Comment:
span name should be `{operation name} {destination}` - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#span-name. 

I.e. `publish` stays (as rabbit-friendly operation name), but the order has changed and should now be `publish {routingKey}`

Or if  you want to call it `basic_publish` instead of `publish` or anything else (language-agnostic) - go for it - happy to update otel semconv with your preference.